### PR TITLE
Plugin Scheduler: Check for file mod capabilities

### PR DESCRIPTION
**Pull request by vishnugopal**
Related to https://github.com/Automattic/dotcom-forge/issues/5874

## Proposed Changes

* Checks the `/v1.2/plugins` API's `file_mod_capabilities` to verify if we can schedule plugins.
* This check applies to both the list and the create pages.
* We disable buttons and display appropriate notices if we can't schedule plugins.
* We start with the assumption that schedule creation is possible, and wait for the API to return to show disable states if necessary. This seemed to be the safest thing to do.

## Testing Instructions

* Apply this PR: https://github.com/Automattic/jetpack/pull/36238 and this diff: D141065-code by following instructions there.
* Go to `/plugins/scheduled-updates/site` 
* Go to `/plugins/scheduled-updates/create/site` 
* Go to `/plugins/scheduled-updates/edit/site/id` 
* You should see the default state.

Now, to test a bad result, add a line here: https://github.com/Automattic/jetpack/pull/36238/files#diff-16bb61901772bfecbc9020e4b205240d10a347c6f5e6ac61fe795186bcc4d997R234: 

```
		$reasons_can_not_modify_files['is_sub_site'] = __( 'The site is not the main network site', 'jetpack-scheduled-updates' );

```

* Repeat the above checks to see the new UI.

CleanShot 2024-03-05 at 14 56 09
CleanShot 2024-03-05 at 14 59 12
CleanShot 2024-03-07 at 13 34 52

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by bogiii** [comment]
Screenshot 2024-03-04 at 11 37 29

Following Figma, the disabled button should be `variant="secondary"` to match the design.

**Comment by vishnugopal** [comment]
> Following Figma, the disabled button should be `variant="secondary"` to match the design.

Fixed @bogiii thanks!

File: client/blocks/plugins-update-manager/schedule-create.tsx [hunk 1]
```diff
@@ -34,9 +36,15 @@ export const ScheduleCreate = ( props: Props ) => {
...
 			</CardBody>
 			<CardFooter>
-				<Button form="schedule" type="submit" variant="primary">
+				<Button form="schedule" type="submit" variant="primary" disabled={ ! canCreateSchedules }>
 					Create
 				</Button>
+				{ ! canCreateSchedules && (
+					<Text as="p">
+						<Icon className="icon-info" icon={ warning } size={ 16 } />
+						This site cannot auto-update plugins.
```
Comment (ouikhuan) [inline client/blocks/plugins-update-manager/schedule-create.tsx]: I feel like we might want to emphasize the word "schedule" here. Perhaps something like "This site is unable to schedule auto-updates for plugins." but not a blocker here. What do you think?

Comment (obenland) [reply client/blocks/plugins-update-manager/schedule-create.tsx]: Let's also add why the site can't updates so users have an understanding how to fix it

Comment (vishnugopal) [reply client/blocks/plugins-update-manager/schedule-create.tsx]: I agree, but let's leave it to the copy editor? None of the strings have been copy-checked, so let's consider this a placeholder for now?

Comment (obenland) [reply client/blocks/plugins-update-manager/schedule-create.tsx]: I'm generally a proponent of ship and iterate, but for some reason user-facing strings feel different for me. Maybe I've seen them fall through the cracks too many times.

Comment (vishnugopal) [reply client/blocks/plugins-update-manager/schedule-create.tsx]: I have changed this to `This site is unable to schedule auto-updates for plugins.`

@obenland Could you suggest how users can fix these? From my general googling of the problem the possible causes & solutions are many, so perhaps we should write a separate support doc ourselves.

Comment (obenland) [reply client/blocks/plugins-update-manager/schedule-create.tsx]: In the endpoint you'll write, let's return a WP error, [similar to what Jetpack does in the plugins endpoint](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/json-endpoints/jetpack/class.jetpack-json-api-plugins-endpoint.php#L298-L341). We can probably even re-use those messages.

Comment (vishnugopal) [reply client/blocks/plugins-update-manager/schedule-create.tsx]: New endpoint here: D140894-code I decided not to return `WP_Error` as it doesn't seem to be a client or a server error, just an application error state.

Comment (vishnugopal) [reply client/blocks/plugins-update-manager/schedule-create.tsx]: Adapted to use PR here: https://github.com/Automattic/jetpack/pull/36238

**Review by TimBroddin — APPROVED** [review]
Nice job!

**Pull request merged**